kclvm/tools/src/query/override.rs

```diff
@@ -138,44 +274,107 @@ impl<'ctx> MutSelfMutWalker<'ctx> for OverrideTransformer {
             self.walk_expr(&mut config_entry.node.value.node);
         }
     }
+
+    fn walk_if_stmt(&mut self, _: &'ctx mut ast::IfStmt) {
+        // Do not override AssignStmt in IfStmt
+    }
+    fn walk_schema_stmt(&mut self, _: &'ctx mut ast::SchemaStmt) {
+        // Do not override AssignStmt in SchemaStmt
+    }
+    fn walk_lambda_expr(&mut self, _: &'ctx mut ast::LambdaExpr) {
+        // Do not override AssignStmt in LambdaExpr
+    }
 }
 
 impl OverrideTransformer {
-    pub(crate) fn _get_schema_config_field_paths(
-        &mut self,
-        schema_expr: &mut ast::SchemaExpr,
-    ) -> (Vec<String>, Vec<String>) {
-        if let ast::Expr::Config(config_expr) = &mut schema_expr.config.node {
-            self._get_config_field_paths(config_expr)
-        } else {
-            (vec![], vec![])
+    /// Lookup schema config all fields and replace if it is matched with the override spec,
+    /// return whether is found a replaced one.
+    fn lookup_config_and_replace(&mut self, config_expr: &mut ast::ConfigExpr) -> bool {
+        let paths = get_attr_paths_from_config_expr(config_expr);
+        match paths.iter().position(|r| r == &self.field_path) {
+            Some(pos) => {
+                let path = &paths[pos];
+                // Split a path into multiple parts. `a.b.c` -> ["a", "b", "c"]
+                let parts = path.split('.').collect::<Vec<&str>>();
+                self.replace_config_with_path_parts(config_expr, &parts);
+                true
+            }
+            None => false,
         }
     }
-    pub(crate) fn _get_config_field_paths(
+
+    /// Replace AST config expr with one part of path.
+    fn replace_config_with_path_parts(
         &mut self,
-        config: &mut ast::ConfigExpr,
-    ) -> (Vec<String>, Vec<String>) {
-        let mut paths = vec![];
-        let mut paths_with_id = vec![];
-        for entry in config.items.iter_mut() {
-            let (mut _paths, mut _paths_with_id) = self._get_key_value_paths(&mut entry.node);
-            paths.append(&mut _paths);
-            paths_with_id.append(&mut &mut _paths_with_id);
+        config_expr: &mut ast::ConfigExpr,
+        parts: &[&str],
+    ) {
+        // Do not replace empty path parts and out of index parts on the config expression.
+        if parts.is_empty() {
+            return;
+        }
+        let part = parts[0];
+        let mut delete_index_set = HashSet::new();
+        // Loop all entries in the config expression and replace.
+        for (i, item) in config_expr.items.iter_mut().enumerate() {
```

Comment:
Pls add comments on why we need a loop here.

Reply:
> Pls add comments on why we need a loop here.

Have added more commennts